geps/gep-3779/index.md

```diff
@@ -241,23 +241,518 @@ spec:
+
+```yaml
+targetRefs:
+- Kind: Pod
+  Selector: {}
+```
+
+This provides a clearer separation between workloads and gateways. Gateways that happen to be Pods in the cluster, will also get selected. However they can be excluded with the right `matchExpression` if desired.
+
+Below is a pseudo example:
+
+```yaml
+targetRefs:
+- Kind: Pod
+  Selector:
+    matchExpression:
+    - { key: "purpose", operator: NotIn, values: ["gateway"] }
+```
+
+##### **Enhanced Discoverability with `gwctl`**
+
+A key challenge with `LabelSelector` is the loss of discoverability. It’s easier to see which policies target a `Service` but difficult to determine which policies might affect a specific pod.
+
+To address this, **investment in tooling is required.** Specifically, the `gwctl` CLI tool should be enhanced to provide insights such as:
+
+```sh
+gwctl describe pods pod1 
+
+...
+InheritedPolicies:
+  Type                   Name                                 Target Kind   Target Name/Expression
+  ----                   ----                                 -----------   -----------
+  AuthorizationPolicy  demo-authz-policy-on-payment-pods      Pod           Selector={foo: a, bar: 2}
+...
+
+# List all AuthorizationPolicy resources in json format in the active namespace
+gwctl get authorizationPolicy -o json
+
+```
+
+Without dedicated tooling, the `LabelSelector` approach could significantly degrade the user experience and observability.
+
+#### Recommended Option - Hybrid TargetRef and VAP
+
+Adding only the recommendation below, please see [https://docs.google.com/document/d/1CeagBnHDPbzpYAxBmtJqTshxAW8l-aRPwvwgAGbnv2I/edit?tab=t.0](https://docs.google.com/document/d/1CeagBnHDPbzpYAxBmtJqTshxAW8l-aRPwvwgAGbnv2I/edit?tab=t.0) for a comprehensive comparison for design options for TargetRef.
+
+Create one unified authorization policy API with implementation-specific validation using Validating Admission Policies (VAP). Additionally introducing a new `enforcementLevel` enum to simplify validation and enable the user to clarify intent.
+
+```go
+
+// EnforcementLevel defines the scope at which an AuthorizationPolicy is enforced.
+//
+// There are two enforcement levels:
+//
+//   - NETWORK: Enforces the policy at the network layer (L4), typically at
+//     network proxies or gateway dataplanes. Only supports attributes available
+//     at connection time (e.g., source identity, port). Recommended for broad,
+//     coarse-grained access controls.
+//
+//   - APPLICATION: Enforces the policy at the application layer (L7), typically at
+//     HTTP/gRPC-aware sidecars or L7 proxies. Enables fine-grained authorization
+//     using protocol-specific attributes (e.g., HTTP paths, methods).
+//
+// This field clarifies policy intent and informs where enforcement is expected
+// to happen. It also enables implementation-specific validation and behavior.
+//
+// +kubebuilder:validation:Enum=NETWORK;APPLICATION
+type EnforcementLevel string
```

Comment:
Please include this in API spec as well. If we add it in future, it may bring some breaking changes.